core/src/main/scala/org/bykn/bosatsu/Package.scala

```diff
@@ -152,6 +152,14 @@ object Package {
           .leftMap { badPaths =>
             badPaths.map(PackageError.CircularType(p, _))
           }
+      /*
+       * Check that all recursion is allowable
+       */
+      val defRecursionCheck: ValidatedNel[PackageError, Unit] =
```

Comment:
we aren't actually using this... :/ nice work man... Add a test that shows the omission, then fix.